### PR TITLE
Auth refactor

### PR DESCRIPTION
**Pull request by neb42**
Jira ticket: JIRA-CORE24-#

## What have you implemented?
The Azure authorisation provider didn't allow the SCIM provisioning to authenticate. This PR removes the Azure auth provider and adds the auth handling to the express app.

Okta SCIM wants to use SAML or their own proprietary auth mechanism to authenticate. To use OIDC we have two apps in Okta. One is for SCIM that handles the user provisioning. The other is for OIDC that handles user auth. Access to both applications is handled through a single user group. Adding a user to this group will trigger the SCIM workflow and allow them to login.

I suspect there are issues here and it will need some future work (possibly to do with the frontend and redirect workflows).

This has caused the local oidc provider to not work. I am working on a fix for this but in the meantime use a dev Okta account.

* Setup passport.js
* Remove Azure auth
* Fix various things, eg. db connections, migrations, libs in docker image
* Move me controller into user controller
* Refactor auth middleware to use passport.js

## How to test it?
On core.app.nrc-dev.no, add/remove/change users and check that they are updated in core.app.nrc-dev.no/api/user

## Extra/next steps
* There is a postgres setting that I am unsure about. Needs DevOps input.
* Local database seeding isn't setup. Either we have it pull from Okta or wait until I have a working local oidc.
* I think we should move the user logic to a `user-engine` lib.

### CODE REVIEW

File: apps/core-api/src/controllers/user.controller.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  }
+});
+
+// TODO: Move to models lib
+const paginationSchema = z.object({
+  startIndex: z.number().int().positive().default(0),
+  limit: z.number().int().positive().default(100),
+});
+
+router.get('/user', async (req, res, next) => {
```
Comment (internetti) [inline apps/core-api/src/controllers/user.controller.ts:27]: I'd be for plural urls

File: apps/core-api/src/middleware/oidc.middleware.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+import passport from 'passport';
+import { Strategy as OidcStrategy } from 'passport-openidconnect';
+
+import * as UserService from '../services/user.service';
+import { getServerConfig } from '../config';
+
+export const oidc = () => {
+  const config = getServerConfig();
+
+  const app = Router();
```
Comment (internetti) [inline apps/core-api/src/middleware/oidc.middleware.ts:12]: detail, but I find it confusing to name this app, like in main.ts when it's only a router. I'd name it maybe oidcRouter

**Review by internetti — COMMENTED** [review]
getting this error on api serve:
TypeError: OpenIDConnectStrategy requires a clientID option

need to add/change something on my end? or does it not make sense to run this locally at all right now, even with the dev okta in the env file?

File: apps/core-api/src/controllers/user.controller.ts (hunk 1 above)
Comment (neb42) [reply apps/core-api/src/controllers/user.controller.ts:27]: I'm for singular. Happy to leave it for now and let David cast the deciding vote when he's back.

**Comment by neb42** [comment]
> getting this error on api serve: TypeError: OpenIDConnectStrategy requires a clientID option
> 
> need to add/change something on my end? or does it not make sense to run this locally at all right now, even with the dev okta in the env file?

@internetti this sounds like an issue in your .env. The example .env doesn't have the client id and secret set. They can be pulled from dev okta.

File: apps/core-api/src/controllers/user.controller.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import { Router } from 'express';
+import { z } from 'zod';
+
+import * as UserService from '../services/user.service';
+
+const router = Router();
+
+router.get('/me', async (req, res, next) => {
```
Comment (dfigueroa87) [inline apps/core-api/src/controllers/user.controller.ts:8]: A bit late to the party, but I think it'd make sense to put this under `/users/me`

**Pull request merged**